### PR TITLE
[Task#48455] Unit Test StatisticController & Notifications

### PR DESCRIPTION
**Pull request by khieutrung12**
## Related Tickets:

-   [#48455](https://edu-redmine.sun-asterisk.vn/issues/48445)

## WHAT this PR does?

- Unit test StatisticController
- Unit test Notifications
- Refactor StatisticController

## Notes Impacted Areas:

-   N/A.

## Attachments:

Screenshot from 2022-04-18 10-03-33

### CODE REVIEW

**Comment by khieutrung12** [comment]
ready

File: tests/Unit/Http/Controllers/StatisticControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        $this->assertJson(json_encode($value), json_encode($response->getData()));
+    }
+
+    public function testSelectMonthOrderFailNoData()
+    {
+        $get = [];
+        $get = [
+            'year' => '',
+            'month' => ''
+        ];
```
Comment (ghost) [inline tests/Unit/Http/Controllers/StatisticControllerTest.php]:
```suggestion
        $get = [
            'year' => '',
            'month' => '',
        ];
```

File: tests/Unit/Http/Controllers/StatisticControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        $response = $this->controller->selectMonthOrder($request);
+        $this->assertJson(json_encode($value), json_encode($response->getData()));
+    }
+
+    public function testSelectMonthOrderFail()
+    {
+        $get = [];
+        $get = [
+            'year' => '2021',
+            'month' => '01'
```
Comment (ghost) [inline tests/Unit/Http/Controllers/StatisticControllerTest.php]:
```suggestion
        $get = [
            'year' => '2021',
            'month' => '01',
```

File: tests/Unit/Http/Controllers/StatisticControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        $this->assertJson(json_encode($value), json_encode($response->getData()));
+    }
+
+    public function testSelectMonthOrderSuccess()
+    {
+        $get = [];
+        $get = [
+            'year' => '2021',
+            'month' => '01'
+        ];
```
Comment (ghost) [inline tests/Unit/Http/Controllers/StatisticControllerTest.php]:
```suggestion
        $get = [
            'year' => '2021',
            'month' => '01',
        ];
```

File: resources/js/statistic_revenue.js [hunk 4]
```diff
@@ -36,6 +38,9 @@ $(function () {
                     },
                 }
             });
+        }else{
```
Comment (ghost) [inline resources/js/statistic_revenue.js]:
```suggestion
        } else {
```

File: app/Http/Controllers/StatisticController.php [hunk 5]
```diff
@@ -58,47 +61,79 @@ public function selectYearRevenue(Request $request)
...
+        if ($request->year && $request->month) {
+            $data = [];
+            $totalOrder = "";
+            $year = $request->year;
+            $numberMonth = $request->month;
+            $month = date("F", mktime(0, 0, 0, $numberMonth, 1));
+            $nextNumberMonth = $request->month + 1;
+            $nextMonth = date("F", mktime(0, 0, 0, $nextNumberMonth, 1));
+            //Lấy ra các ngày thứ 2 trong tháng
+            $firstDayOfWeek = new \DatePeriod(
```
Comment (ghost) [inline app/Http/Controllers/StatisticController.php]: use DatePeriod trên cùng nha

Comment (khieutrung12) [reply app/Http/Controllers/StatisticController.php]: @manhtd-1083 vâng ạ

**Comment by khieutrung12** [comment]
ready

**Review by DTHNhung — APPROVED** [review]

File: app/Repositories/Order/OrderRepository.php [hunk 6]
```diff
@@ -85,4 +85,14 @@ public function getOrdersOnWeek($fromDate, $toDate)
...
     }
+
+    public function countOrderByYear($year)
+    {
+        $data = $this->model->whereYear('updated_at', '=', $year)->get();
+        if (count($data) > 0) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (huyenttm-0901) [inline app/Repositories/Order/OrderRepository.php]:
```suggestion
       return count($data) > 0;
```

**Comment by khieutrung12** [comment]
ready

**Pull request merged**